Sidebar improvements

elbotho opened the pull request:
- spinner color
- less updates
- scrollTop fixed

elbotho committed:
new(SearchBar): Spinner also gray

elbotho committed:
fix(ResultList): only update resultless if it changed

Increases overall performance especially when zooming the map

lazy loading and only updating when the first few entries change might
be even faster.

elbotho committed:
new(Sidebar): less updates & less scrolling

I hope I got all the right props. Not beautiful at all, but seems
faster.
Also scrolling should now work on old androids as well and should only
happen when needed.

elbotho committed:
new(BusinessCard): also shorten long mail addresses

closes #469

elbotho committed:
new(BusinessCard): Show category over title in ENTRY

elbotho committed:
fix(ResultList): pure(ResultListEntry)

elbotho committed:
new(EntryDetail): Add date of last change and version number

david-ziegler reviewed (commented):
is it on purpose that "last edit" is always in English. It doesn't really hurt, since it's only power-user-information or so. Just asking.

elbotho committed:
new(EntryDetails): Report mailto-link

closes #165

elbotho committed:
new(Ratings): show url-like stings as links

closes #264

elbotho committed:
new(EntryForm): Show all pins when editing

- resets search string so it shows all pins again.
- if a marker is set, the rest of the pins are set to a lot opacity
closes #356

elbotho committed:
fix(Flower): don't show tooltips in ResultList

closes #476

elbotho committed:
fix(Sidebar): also update when ratings change

elbotho committed:
new(BusinessCard): link phone numbers

closes #123

elbotho committed:
fix(*): use global styles again, update styled-components

elbotho committed:
fix(Widget): remove styl dependency

closes #475

elbotho committed:
fix(styled): cleanup

elbotho committed:
new(BusinessCard): route planner redirects to native apps on iOS and android

uses apple maps on iOS and GeoURI on android. Graphhopper as fallback
and on desktops.
closes #457

elbotho committed:
chore(*): removing stylus dep

elbotho committed:
new(Sidebar): on small screens, hide sidebar on map touch/click

elbotho committed:
fix(Sidebar): swipe actions for sidebar again

- `react-swipeable` instead of `react-swipe-component`
- no animation but should work
more testing would be great but should degradate ok.

elbotho committed:
new(Sidebar): Back to search results on mapclick

closes #481
does not close on swipe/mapmove on purpose

elbotho committed:
Merge remote-tracking branch 'flosse/master' into sidebar-improvements
